test/unix/test_lwt_bytes.ml

```diff
@@ -59,6 +59,42 @@ let gen_buf n =
   let () = Lwt_bytes.fill buf 0 n '\x00' in
   buf
 
+let test_mincore buff_len offset n_states =
+  let test_file = "bytes_mincore_write" in
+  Lwt_unix.openfile test_file [O_RDWR;O_TRUNC; O_CREAT] 0o666
+  >>= fun fd ->
+  let buf_write = gen_buf buff_len in
+  Lwt_bytes.write fd buf_write 0 buff_len
+  >>= fun _n ->
+  Lwt_unix.close fd
+  >>= fun () ->
+  let fd = Unix.openfile test_file [O_RDONLY] 0 in
+  let shared = false in
+  let size = buff_len in
+  let buffer = Lwt_bytes.map_file ~fd ~shared ~size () in
+  let states = Array.make n_states false in
+  let () = Lwt_bytes.mincore buffer offset states in
+  Lwt.return states
+
+let test_wait_mincore buff_len offset =
+  let test_file = "bytes_mincore_write" in
+  Lwt_unix.openfile test_file [O_RDWR;O_TRUNC; O_CREAT] 0o666
+  >>= fun fd ->
+  let buf_write = gen_buf buff_len in
+  Lwt_bytes.write fd buf_write 0 buff_len
+  >>= fun _n ->
+  Lwt_unix.close fd
+  >>= fun () ->
+  let fd = Unix.openfile test_file [O_RDONLY] 0 in
+  let shared = false in
+  let size = buff_len in
+  let buffer = Lwt_bytes.map_file ~fd ~shared ~size () in
+  Lwt_bytes.wait_mincore buffer offset
```

Comment:
I think we should check that `mincore` returns `false` for the page being waited for. If we can't observe `false`, the `wait_mincore` tests aren't telling us anything, except that bounds checking is working.

If we can't get `false`, I suggest to rename the `wait_mincore` success test at the end to something like `wait_mincore correct bounds`, and leave a comment explaining why the test isn't testing the function "truly."

Reply:
done